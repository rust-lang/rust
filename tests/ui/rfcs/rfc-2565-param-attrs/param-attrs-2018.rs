//@ edition:2018

trait Trait2015 { fn foo(#[allow(C)] i32); }
//~^ ERROR expected one of `:`, `@`, or `|`, found `)`

fn main() {}
