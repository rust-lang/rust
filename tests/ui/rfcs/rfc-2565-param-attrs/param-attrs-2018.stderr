error: expected one of `:`, `@`, or `|`, found `)`
  --> $DIR/param-attrs-2018.rs:3:41
   |
LL | trait Trait2015 { fn foo(#[allow(C)] i32); }
   |                                         ^ expected one of `:`, `@`, or `|`
   |
   = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
help: if this is a `self` type, give it a parameter name
   |
LL | trait Trait2015 { fn foo(#[allow(C)] self: i32); }
   |                                      +++++
help: if this is a parameter name, give it a type
   |
LL | trait Trait2015 { fn foo(#[allow(C)] i32: TypeName); }
   |                                         ++++++++++
help: if this is a type, explicitly ignore the parameter name
   |
LL | trait Trait2015 { fn foo(#[allow(C)] _: i32); }
   |                                      ++

error: aborting due to 1 previous error

